connectivity/mqtt/pub_utils.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "pub_utils.h"
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "utils/logger_helper.h"
+
+#define MQTT_PUB_LOGGER "mqtt-pub"
+static logger_id_t mqtt_pub_logger_id;
+
+void mqtt_pub_logger_init() { mqtt_pub_logger_id = logger_helper_enable(MQTT_PUB_LOGGER, LOGGER_DEBUG, true); }
+
+int mqtt_pub_logger_release() {
+  logger_helper_release(mqtt_pub_logger_id);
+  if (logger_helper_destroy() != RC_OK) {
+    log_critical(mqtt_pub_logger_id, "[%s:%s] Destroying logger failed %s.\n", __func__, __LINE__, MQTT_PUB_LOGGER);
+    return EXIT_FAILURE;
+  }
+
+  return 0;
+}
+
+mosq_retcode_t publish_message(struct mosquitto *mosq, mosq_config_t *cfg, int *mid, const char *topic, int payloadlen,
+                               void *payload, int qos, bool retain) {
+  if (mosq == NULL || cfg == NULL) {
+    log_error(mqtt_pub_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_TA_NULL");
+    return SC_MQTT_NULL;
+  }
+
+  if (cfg->general_config->protocol_version == MQTT_PROTOCOL_V5 && cfg->pub_config->first_publish == false) {
+    return mosquitto_publish_v5(mosq, mid, NULL, payloadlen, payload, qos, retain, cfg->property_config->publish_props);
```

Comment:
You shall address the use of `_v5`. At least, explain it.

Reply:
We choose to use `MQTT_PROTOCOL_V5`, since the client which can send and receive message (`client_rr` in `mosquitto`) supports only `MQTT_PROTOCOL_V5`.
Should I